### PR TITLE
🌱 Add v1beta2 conditions to ClusterClass

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/11105

### CODE REVIEW

**Comment by sbueringer** [comment]
/hold
Needs a bit more refinement

File: internal/controllers/clusterclass/clusterclass_controller_status.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+				clusterv1.ClusterClassOutdatedRefVersionsReason,
+				clusterv1.ConditionSeverityWarning,
+				strings.Join(msg, ", "),
+			),
+		)
+		v1beta2conditions.Set(clusterClass, metav1.Condition{
+			Type:    clusterv1.ClusterClassRefVersionsUpToDateV1Beta2Condition,
+			Status:  metav1.ConditionFalse,
+			Reason:  clusterv1.ClusterClassRefVersionsNotUpToDateV1Beta2Reason,
+			Message: strings.Join(msg, ", "),
```
Comment (fabriziopandini) [inline internal/controllers/clusterclass/clusterclass_controller_status.go:69]: At least for v1beta2, let's make this a list (use * as prefix and join by \n)

Comment (sbueringer) [reply internal/controllers/clusterclass/clusterclass_controller_status.go:69]: Changed it for both just simpler

**Comment by sbueringer** [comment]
@fabriziopandini PTAL

**Review by fabriziopandini — COMMENTED** [review]
Nice!
/lgtm
/approve

**Pull request merged**